diff --git a/flake.nix b/flake.nix
index c6645ad..ffe0d4e 100644
--- a/flake.nix
+++ b/flake.nix
@@ -7,7 +7,10 @@
   inputs.luispkgs.url = "github:Luis-Hebendanz/nixpkgs/luispkgs";
   inputs.unstable.url = "github:NixOS/nixpkgs/nixos-unstable";
   inputs.master.url = "github:NixOS/nixpkgs/master";
-
+  inputs.clan-core = {
+    url = "git+https://git.clan.lol/clan/clan-core";
+    inputs.nixpkgs.follows = "nixpkgs";
+  };
 
 
   inputs.retiolum = {
@@ -21,51 +24,57 @@
   # inputs.nixpack.url  ="github:dguibert/nixpack/dg/flake";
 
 
-  outputs = { self, nixpkgs, retiolum, nur, luispkgs, unstable, master, chrome-pwa }: {
-    # replace 'qubasa-desktop' with your hostname here.
-    nixosConfigurations.qubasa-desktop = nixpkgs.lib.nixosSystem rec {
-      system = "x86_64-linux";
-      modules = [
-        ./configuration.nix
-        chrome-pwa.nixosModule
+  outputs = { self, clan-core, nixpkgs, retiolum, nur, luispkgs, unstable, master, chrome-pwa }: {
+    nixosConfigurations = clan-core.lib.buildClan {
+      directory = ./.;
+      # replace 'qubasa-desktop' with your hostname here.
+      machines = {
+        qubasa-desktop = {
+          #nixpkgs.pkgs = nixpkgs.legacyPackages.x86_64-linux;
+          nixpkgs.hostPlatform = "x86_64-linux";
+          imports = [
+            ./configuration.nix
+            chrome-pwa.nixosModule
 
-        { nixpkgs.overlays = [ nur.overlay ]; }
+            { nixpkgs.overlays = [ nur.overlay ]; }
 
-        ({ ... }: {
-          # Set nix-channel to nix flake registry
-          nix.nixPath = let path = toString ./.; in [ "repl=${path}/repl.nix" "nixpkgs=${self.inputs.nixpkgs}" ];
+            ({ ... }: {
+              # Set nix-channel to nix flake registry
+              nix.nixPath = let path = toString ./.; in [ "repl=${path}/repl.nix" "nixpkgs=${self.inputs.nixpkgs}" ];
 
-          nixpkgs.config = {
-            allowUnfree = true;
-            packageOverrides = pkgs:
-              {
-
-                retiolum = retiolum;
-                luis = import luispkgs
-                  {
-                    system = "x86_64-linux";
-                    config = {
-                      allowUnfree = true;
-                    };
-                  };
-                unstable = import unstable
-                  {
-                    system = "x86_64-linux";
-                    config = {
-                      allowUnfree = true;
-                    };
-                  };
-                master = import master
+              nixpkgs.config = {
+                allowUnfree = true;
+                packageOverrides = pkgs:
                   {
-                    system = "x86_64-linux";
-                    config = {
-                      allowUnfree = true;
-                    };
+
+                    retiolum = retiolum;
+                    luis = import luispkgs
+                      {
+                        system = "x86_64-linux";
+                        config = {
+                          allowUnfree = true;
+                        };
+                      };
+                    unstable = import unstable
+                      {
+                        system = "x86_64-linux";
+                        config = {
+                          allowUnfree = true;
+                        };
+                      };
+                    master = import master
+                      {
+                        system = "x86_64-linux";
+                        config = {
+                          allowUnfree = true;
+                        };
+                      };
                   };
               };
-          };
-        })
-      ];
+            })
+          ];
+        };
+      };
     };
   };
 }
